prompt_manager/ipromptmanager.py

```diff
@@ -9,7 +9,11 @@ class IPromptManager(ABC):
     It generates specific prompts for a given Agent (Code Generation or Analysis Suggestion and Interpretation) based on the current conversation and the LLM type.
     """
 
-    @abstractmethod
+    prompt_generators = {
```

Comment:
Convert to abstract methods

Reply:
Since it is not clear for me whether we should use provided here method or abstrac methods and you are suggesting that abstract methods are much clearer I will change it to abstract methods